mobile/lib/modules/home/ui/asset_grid/immich_asset_grid_view.dart

```diff
@@ -223,6 +233,22 @@ class ImmichAssetGridViewState extends State<ImmichAssetGridView> {
         : RefreshIndicator(onRefresh: widget.onRefresh!, child: child);
   }
 
+  void _scrollToDate(DateTime? date) {
+    if (date == null) {
+      return;
+    }
+    final index = widget.renderList.elements.indexWhere(
+      (e) =>
+          e.date.year == date.year &&
+          e.date.month == date.month &&
+          e.date.day == date.day,
+    );
+    if (index != -1) {
+      // Not sure why the index is shifted, but it works. :3
+      _scrollToIndex(index + 1);
```

Comment:
This is rather odd, but if it works, it works! I wonder if we shouldn't double check the position if it's the last asset in the list to avoid an overflow?

Reply:
After some testing, it seems that scrolling to an index too high just gets silently swallowed and the list scrolls to the last item anyways, but I added a check anyhow.